plenum/server/future_primaries_batch_handler.py

```diff
@@ -33,74 +19,47 @@ class FuturePrimariesBatchHandler(BatchRequestHandler):
-        node_txn_count = 0
-        last_state = None
-        if len(self.node_states) == 0:
-            last_state = self.create_node_state_from_current_node()
-        else:
-            last_state = copy.deepcopy(self.node_states[-1])
-
+        node_reg = list(self.node.nodeReg.keys())
+        node_ids = copy.deepcopy(self.node.nodeIds)
         for digest in three_pc_batch.valid_digests:
             if digest not in self.node.requests:
                 raise LogicError('Request is absent when it is applying')
             request = self.node.requests[digest].request
             if request.operation.get(TXN_TYPE) == NODE \
                     and request.operation.get(DATA).get(SERVICES) is not None:
-                node_txn_count += 1
                 node_nym = request.operation.get(TARGET_NYM)
                 node_name = request.operation.get(DATA).get(ALIAS)
-                curName = last_state.node_ids.get(node_nym)
+                curName = node_ids.get(node_nym)
                 if curName is None:
-                    last_state.node_ids[node_nym] = node_name
+                    node_ids[node_nym] = node_name
                 elif curName != node_name:
                     raise LogicError("Alias inconsistency")
 
                 serv = request.operation.get(DATA).get(SERVICES)
-                if VALIDATOR in serv and node_name not in last_state.node_reg:
-                    last_state.node_reg.append(node_name)
-                elif serv == [] and node_name in last_state.node_reg:
-                    last_state.node_reg.remove(node_name)
-
-                count = self.get_required_number_of_instances(len(last_state.node_reg))
-                if last_state.number_of_inst != count:
-                    last_state.number_of_inst = count
-                    new_validators = TxnPoolManager.calc_node_names_ordered_by_rank(last_state.node_reg,
-                                                                                    last_state.node_ids)
-                    last_state.primaries = self.node.primaries_selector.select_primaries(view_no=self.node.viewNo,
-                                                                                         instance_count=last_state.number_of_inst,
-                                                                                         validators=new_validators)
-
-        # We will save node state at every pool batch, so we could revert it correctly
-        self.node_states.append(last_state)
-        three_pc_batch.primaries = last_state.primaries
-        return last_state.primaries
+                if VALIDATOR in serv and node_name not in node_reg:
+                    node_reg.append(node_name)
+                elif serv == [] and node_name in node_reg:
+                    node_reg.remove(node_name)
+
+        count = self.get_required_number_of_instances(len(node_reg))
+
+        number_of_inst = count
```

Comment:
Why don't name `count` a `number_of_inst ` to avoid useless `number_of_inst = count`?